### PR TITLE
0box nft state/info system test

### PR DESCRIPTION
**Pull request by Manali-Jain-squareops**

### CODE REVIEW

**Comment by Manali-Jain-squareops** [comment]
NFT State (PUT Request)
image

File: tests/api_tests/0box_nft_state_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestNftState(testSetup *testing.T) {
+	t := test.NewSystemTest(testSetup)
+	t.Parallel()
+
+	firebaseToken := authenticateWithFirebase(t, zboxClient.DefaultPhoneNumber)
+
+	t.RunSequentially("Create a wallet with valid phone number should work", func(t *test.SystemTest) {
```
Comment (Kishan-Dhakan) [inline tests/api_tests/0box_nft_state_test.go:16]: Please change the description from to `Create a wallet with valid phone number should work` to something more relevant like "Posting NFT state with valid formdata should work"

File: tests/api_tests/0box_nft_state_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		NftState, response, err := zboxClient.PostNftState(t,
+			firebaseToken.IdToken,
+			csrfToken,
+			zboxClient.DefaultPhoneNumber,
+		)
+		require.NoError(t, err)
+		require.Equal(t, 201, response.StatusCode())
+		require.NotNil(t, NftState)
+	})
+
```
Comment (Kishan-Dhakan) [inline tests/api_tests/0box_nft_state_test.go:44]: Please remove the white spaces to solve lint issue.

File: tests/api_tests/0box_nft_state_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		require.Equal(t, walletName, zboxWallet.Name, "Wallet name does not match expected")
+
+		NftState, response, err := zboxClient.PostNftState(t,
+			firebaseToken.IdToken,
+			csrfToken,
+			zboxClient.DefaultPhoneNumber,
+		)
+		require.NoError(t, err)
+		require.Equal(t, 201, response.StatusCode())
+		require.NotNil(t, NftState)
```
Comment (Kishan-Dhakan) [inline tests/api_tests/0box_nft_state_test.go:42]: Please make some stronger assertions here, other than notNil. For example, allocation id should be correct, collection id, so on.

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
This only has one case, valid PUT NFT State case. Please check the issue https://github.com/0chain/system_test/issues/513. and add more coverage listed there, for example, "put nft state with invalid allocation id should fail", and other endpoints/methods like NFT Info, GET NFT State/Info, so on

**Review by stewartie4 — CHANGES_REQUESTED** [review]
Some missing coverage, please see the ticket description 
The list there also isn't exhaustive so feel free to think of other test cases by playing around with the APIs

**Comment by Manali-Jain-squareops** [comment]
> This only has one case, valid PUT NFT State case. Please check the issue #513. and add more coverage listed there, for example, "put nft state with invalid allocation id should fail", and other endpoints/methods like NFT Info, GET NFT State/Info, so on

still in-progress

File: internal/api/util/client/zbox_client.go [hunk 4]
```diff
@@ -208,3 +208,41 @@ func (c *ZboxClient) ListWalletKeys(t *test.SystemTest, idToken, csrfToken, phon
...
+			"X-App-Client-Key":       "b6d86a895b9ab247b9d19280d142ffb68c3d89833db368d9a2ee9346fa378a05441635a5951d2f6a209c9ca63dc903353739bfa8ba79bad17690fe8e38622e96",
+			"X-App-Client-Signature": "d903d0f57c96b052d907afddb62777a1f77a147aee5ed2b5d8bab60a9319b09a",
+			"X-App-Timestamp":        "1618213324",
+			"X-App-ID-TOKEN":         idToken,
+			"X-App-Phone-Number":     phoneNumber,
+			"X-CSRF-TOKEN":           csrfToken,
+			"X-APP-TYPE":             "chimney",
+		},
+		RequiredStatusCode: 201,
+	}, HttpPUTMethod)
```
[the remark this answers is not in the document]
Comment (Manali-Jain-squareops) [reply internal/api/util/client/zbox_client.go:245]: yes

**Comment by Manali-Jain-squareops** [comment]
NFT INFO POST and PUT Method
image

NFT INFO POST Method
nft-info-POST

nftPostPASS

**Pull request closed without merging**